### ISSUE TITLE
System.Text.Json - JsonSerializer throws NotSupportedException for ignored ConcurrentDictionary property

### ISSUE THREAD

**Issue by realmoriss** [dotnet/corefx#40305]
The following code throws an exception in NET Core 3.0.0-preview8:
```csharp
using System.Collections.Concurrent;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace JsonIgnoreTest
{
    class JsonIgnoreTest
    {
        [JsonIgnore]
        public ConcurrentDictionary<object, object> MyDict { get; set; }
    }

    class Program
    {
        static void Main(string[] args)
        {
            JsonSerializer.Serialize(new JsonIgnoreTest());
        }
    }
}
```

Output:
```
Unhandled exception. System.NotSupportedException: The collection type 'System.Collections.Generic.IDictionary`2[System.Object,System.Object]' is not supported.
   at System.Text.Json.JsonClassInfo.GetElementType(Type propertyType, Type parentType, MemberInfo memberInfo, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.CreateProperty(Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type parentClassType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.AddProperty(Type propertyType, PropertyInfo propertyInfo, Type classType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClass(Type classType)
   at System.Text.Json.WriteStackFrame.Initialize(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at JsonIgnoreTest.Program.Main(String[] args) in C:\Users\user\source\repos\JsonIgnoreTest\JsonIgnoreTest\Program.cs:line 17
```
The problem occurs when the key of the ConcurrentDictionary is not a string.
`ConcurrentDictionary<string, object>`, `IDictionary<object, object>` and `Dictionary<object, object>` do not throw an exception.

**Comment by gudaol** [comment]
lol! Just encountered this today an hour ago!

**Comment by layomia** [comment]
Taking a look at this now.

**Comment by ahsonkhan** [comment]
Re-opening until its fixed for release/3.0

**Comment by layomia** [comment]
Closed for 3.0 by https://github.com/dotnet/corefx/pull/40413.
